dyninstAPI/src/inst-aarch64.C

```diff
@@ -884,14 +884,14 @@ void emitASload(const BPatch_addrSpec_NP *as, Register dest, int stackShift,
         if(ra == 32) {
 	    // Special case where the actual address is store in imm.
 	    // Need to change this for rewriting PIE or shared libraries
-	    insnCodeGen::loadImmIntoReg<long int>(gen, dest, imm);
+	    insnCodeGen::loadImmIntoReg(gen, dest, static_cast<Address>(imm));
 	    return;
 	}
 	else {
 	    restoreGPRtoGPR(gen, ra, dest);
 	}
     } else {
-        insnCodeGen::loadImmIntoReg<long int>(gen, dest, 0);
+        insnCodeGen::loadImmIntoReg(gen, dest, static_cast<Address>(0));
```

Comment:
unnecessary cast

Reply:
Agreed, but I was trying to make it more obvious what was happening at the callsite since the template parameter _should_ have been an `Address`. I guess I could have used `Address{0}`, but I'm going to say it's still better than what was there.